lib/asynchronize.rb

```diff
@@ -1,48 +1,61 @@
 module Asynchronize
   require 'set'
   def self.included(base)
+      
+    ## 
+    # base.class_val  
+    #
+    #  @@asynced_methods     ... methods already asynchronized.
+    #  @@methods_to_async    ... methods that should be asynchronized.
+    #  @@methods_asyncing    ... method that is being asynchronized currently.
+    #    ... Originally used a single value here, but that's not thread safe.
+    #    Though you probably have other problems if you have multiple,
+    #    threads adding methods to your class.
+    #
     base.class_eval do
-      # The methods we have already asynchronized
       @asynced_methods = Set.new
-      # The methods that should be asynchronized.
       @methods_to_async = Set.new
-      # Originally used a single value here, but that's not thread safe.
-      # ...Though you probably have other problems if you have multiple
-      # threads adding methods to your class.
       @methods_asyncing = Set.new
 
-      ##
       # Call to asynchronize a method.
-      #   That method will be added to a list of methods to asynchronize.
-      #   If the method already exists, it will be redefined to an asynchronous
-      #   version. If it does not, method_missing will redefine it when it does.
-      #   If the method is redefined afterwards, method_missing will also
-      #   asynchronize that version.
+      #   Method will be added to a list of methods to asynchronize.
+      #
+      # - If method exists, it will be redefined to an asynchronous version. 
+      # - If method is not defined, method_missing will redefine it when it does.  
```

Comment:
So, it would either read `If method is not defined, method_missing will redefine it when it is.` or `If method does not exist, method_missing will redefine it when it does.`